auth/policies.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package auth
+
+import (
+	"context"
+
+	acl "github.com/ory/keto/proto/ory/keto/acl/v1alpha1"
+)
+
+// Authz represents a authorization service. It exposes
+// functionalities through `auth` to perform authorization.
+type Authz interface {
+	// Authorize checks authorization of the given `subject`. Basically,
+	// Authorize verifies that Is `subject` allowed to `relation` on
+	// `object`. Authorize returns the bool indicating the authorization
+	// of the given subject and the error. For example, the response as:
+	// `false, nil` means that `subject` has not `relation` on the `object`.
+	// Therefore, this response means that incoming request is denied.
+	Authorize(ctx context.Context, subject, object, relation string) (bool, error)
+
+	// AddPolicy creates a policy for the given subject. So that, after
+	// AddPolicy, `subject`has a `relation` on `object`. Returns non-nil
+	// error in case of failures.
+	AddPolicy(ctx context.Context, subject, object, relation string) error
+}
+
+// PolicyAgent facilitates the communication to authorization
+// services and implements Authz functionalities for certain
+// authorization services (e.g. ORY Keto).
+type PolicyAgent interface {
+	// CheckPolicy checks if the subject has a relation on the object.
+	// It returns AuthorizationResult consisting of one boolean which indicates
+	// result of the authorization (true: Allowed, false: Denied) and error.
+	CheckPolicy(ctx context.Context, subject, object, relation string) (AuthorizationResult, error)
+
+	// AddPolicy creates a policy for the given subject. So that, after
+	// AddPolicy, `subject` has a `relation` on `object`. Returns non-nil
+	// error in case of failures.
+	AddPolicy(ctx context.Context, subject, object, relation string) error
+}
+
+// AuthorizationResult shows the result of the authorization check operations.
+// AuthzError field is nil when the operation is allowed. If AuthzError is
+// non-nil, the operation is denied.
+type AuthorizationResult struct {
```

Comment:
Why did you wrap the error? Is it not easier to just return an error? 
For example [here](https://github.com/mainflux/mainflux/pull/1440/files#diff-d60a65301395efdc368c161a4097c96d6f3f1b4500dbcd15de618be7fc9b72a1R72) Looks like a double error for me ?

Reply:
In the beginning (before splitting the PR as indicated [here](https://github.com/mainflux/mainflux/pull/1440#issuecomment-892756436)), I've thought that I might need to return more than one field. However, it is not needed for now. Returning an error is enough.